tests/explainable/test_handlers_explanation.py

```diff
@@ -314,3 +316,236 @@ def test_SplitSubsets_two_layers():
+        trace_connected.trace.nodes["__cause____consequent_Y"]["fn"].log_factor.sum()
+        == 0
+    )
+
+    X_values_rev = trace_reverse.trace.nodes["X"]["value"]
+    if torch.any(X_values_rev == 1.0):
+        assert (
+            trace_reverse.trace.nodes["__cause____consequent_X"]["fn"]
+            .log_factor[1, 0, 0, 0, :]
+            .sum()
+            .exp()
+            == 0.0
+        )
+    else:
+        assert (
+            trace_reverse.trace.nodes["__cause____consequent_X"]["fn"]
+            .log_factor[1, 0, 0, 0, :]
+            .sum()
+            .exp()
+            == 1.0
+        )
+
+    if torch.any(X_values_rev == 0.0):
+        assert (
+            trace_reverse.trace.nodes["__cause____consequent_X"]["fn"]
+            .log_factor[2, 0, 0, 0, :]
+            .sum()
+            .exp()
+            == 0.0
+        )
+    else:
+        assert (
+            trace_reverse.trace.nodes["__cause____consequent_X"]["fn"]
+            .log_factor[2, 0, 0, 0, :]
+            .sum()
+            .exp()
+            == 1.0
+        )
+
+    assert torch.all(
+        trace_reverse.trace.nodes["__cause____consequent_X"]["fn"]
+        .log_factor.sum()
+        .exp()
+        == 0
+    )
+
+
+# X -> Z, Y -> Z
+def model_three_converge():
+    X = pyro.sample("X", dist.Bernoulli(0.5))
+    Y = pyro.sample("Y", dist.Bernoulli(0.5))
+    Z = pyro.sample("Z", dist.Bernoulli(torch.min(X, Y)))
+    return {"X": X, "Y": Y, "Z": Z}
+
+
+# X -> Y, X -> Z
+def model_three_diverge():
+    X = pyro.sample("X", dist.Bernoulli(0.5))
+    Y = pyro.sample("Y", dist.Bernoulli(X))
+    Z = pyro.sample("Z", dist.Bernoulli(X))
+    return {"X": X, "Y": Y, "Z": Z}
+
+
+# X -> Y -> Z
+def model_three_chain():
+    X = pyro.sample("X", dist.Bernoulli(0.5))
+    Y = pyro.sample("Y", dist.Bernoulli(X))
+    Z = pyro.sample("Z", dist.Bernoulli(Y))
+    return {"X": X, "Y": Y, "Z": Z}
+
+
+# X -> Y, X -> Z, Y -> Z
+def model_three_complete():
+    X = pyro.sample("X", dist.Bernoulli(0.5))
+    Y = pyro.sample("Y", dist.Bernoulli(X))
+    Z = pyro.sample("Z", dist.Bernoulli(torch.max(X, Y)))
+    return {"X": X, "Y": Y, "Z": Z}
+
+
+# X -> Y    Z
+def model_three_isolate():
+    X = pyro.sample("X", dist.Bernoulli(0.5))
+    Y = pyro.sample("Y", dist.Bernoulli(X))
+    Z = pyro.sample("Z", dist.Bernoulli(0.5))
+    return {"X": X, "Y": Y, "Z": Z}
+
+
+# X     Y    Z
+def model_three_independent():
+    X = pyro.sample("X", dist.Bernoulli(0.5))
+    Y = pyro.sample("Y", dist.Bernoulli(0.5))
+    Z = pyro.sample("Z", dist.Bernoulli(0.5))
+    return {"X": X, "Y": Y, "Z": Z}
+
+
+@pytest.mark.parametrize("ante_cons", [("X", "Y", "Z"), ("X", "Z", "Y")])
+@pytest.mark.parametrize(
+    "model",
+    [
+        model_three_converge,
+        model_three_diverge,
+        model_three_chain,
+        model_three_complete,
+        model_three_isolate,
+        model_three_independent,
+    ],
+)
+def test_eq_neq_three_variables(model, ante_cons):
```

Comment:
To make this test even more robust, you might consider additionally parametrizing it by an `event_shape` for the model variables.